### ISSUE TITLE
support deprecation

### ISSUE THREAD

**Issue by tmc** [marwan-at-work/protoc-gen-twirpql#3]
Do you plan to support field metadata such as isDeprecated/deprecationReason?

**Comment by marwan-at-work** [comment]
@tmc yes, I plan on pushing deprecated fields to `master` sometime tomorrow :) 

But how would `deprecationReason` be translated from protocol buffers? I don't think the protobuf spec supports deprecation reason right? Protoc-gen-go basically just puts `do not use` as the reason.

**Comment by tmc** [comment]
The two options that come to mind are parsing a comment or introducing a protobuf extension to allow supplying structured metadata.

**Comment by marwan-at-work** [comment]
@tmc I already have a few extensions [here](https://github.com/marwan-at-work/protoc-gen-twirpql/blob/master/options/options.proto) which I haven't written documentation for yet. 

I think we can add a free-form directive option. Such as: 

```proto
message X {
  string y = 1 {
    option (twirpql.options.field) = {
      directives = ["@deprecated(reason: whatever)", "@key"]
    }
  };
}
```

TwirpQL can still support the original deprecated option as well, but it would prioritize the custom inputs
